\documentclass[8pt]{extarticle}
\setlength\parindent{0pt}
\linespread{1}
\pagestyle{empty}

% \usepackage[a4paper, left=15mm, right=15mm, top=15mm, bottom=15mm, showframe]{geometry}
\usepackage[a4paper, left=15mm, right=15mm, top=15mm, bottom=15mm]{geometry}

\usepackage{amsmath}
\usepackage{mathspec}
\setmainfont{FreeSans}[
    Path=fonts/,
    BoldFont=FreeSansBold,
    ItalicFont=FreeSansOblique,
    BoldItalicFont=FreeSansBoldOblique
]
\setmathsfont(Digits,Greek)[Path=fonts/]{FreeSans}
\setmathrm[Path=fonts/]{FreeSans}

\usepackage{graphicx}
\usepackage{textcomp}
\usepackage{enumitem}

\begin{document}

\textbf{\large Figure 1}

\includegraphics[width=8.8cm]{figures/figure_1_spn1_depletion_alternate.pdf}

\vspace{2em}
\textbf{Figure 1. An auxin-inducible degron efficiently depletes Spn1 from chromatin.}
\textbf{a} Western blot showing levels of Spn1 protein in the Spn1 degron strain after 90 minutes of treatment with DMSO (non-depleted) or 25 $\mu$M IAA (Spn1-depleted).
Pgk1 was used as a loading control.
The values below the blots indicate the mean $\pm$ standard deviation of normalized Spn1 abundance quantified from two biological replicates.
\textbf{b} Average Spn1 ChIP enrichment over 3,087 non-overlapping, verified coding genes aligned by TSS in non-depleted and Spn1-depleted conditions.
For each gene, the spike-in normalized ratio of IP over input signal is standardized to the mean and standard deviation of the non-depleted signal over the region, resulting in standard scores that allow genes with varied expression levels to be compared on the same scale.
The solid line and shading are the median and interquartile range of the mean standard score over two replicates.
\textbf{c} Barplot showing total levels of Spn1 on chromatin in non-depleted and Spn1-depleted conditions, estimated from ChIP-seq spike-in normalization factors (see Methods).
The values and error bars for each condition indicate the mean $\pm$ standard deviation of two replicates.
\textbf{d} Scatterplot showing Spn1 ChIP enrichment in Spn1-depleted versus non-depleted conditions for 5,091 verified coding genes.
Enrichment values are the relative log$_2$ enrichment of IP over input.

\newpage

\textbf{\large Figure 2}

\includegraphics[width=18cm]{figures/figure_2_rnaseq_rpb1.pdf}

\vspace{2em}
\textbf{Figure 2. Effects of Spn1 depletion on mRNA levels and RNAPII.}
\textbf{a} Scatterplot showing changes in transcript abundance as measured by RNA-seq upon Spn1 depletion versus non-depleted transcript abundance for 5,091 verified coding transcripts.
Transcripts with significant changes are colored blue, and the relative density of fold-change values is shown in the right panel.
\textbf{b} Top panel: Average Rpb1 ChIP enrichment over 3,087 non-overlapping, verified coding genes in non-depleted and Spn1-depleted conditions.
Spike-in normalized standard scores are calculated as in Fig. 1b.
The solid line and shading are the median and interquartile range of the mean standard score over the two replicates generated from the same chromatin preparations used to perform ChIP-seq for Rpb1-Ser5-P and Rpb1-Ser2-P.
Bottom two panels: Average Rpb1-normalized Rpb1-Ser5-P and Rpb1-Ser2-P ChIP enrichment in non-depleted and Spn1-depleted conditions for the same genes.
The solid line and shading are the median and interquartile range of the mean spike-in normalized ratio over two replicates.
\textbf{c} Scatterplot showing change in transcript abundance upon Spn1 depletion versus change in Rpb1 enrichment for 5,091 verified coding genes.
The line y=x is drawn for comparison, and the Pearson correlation coefficient is shown.
\textbf{d} Spike-in normalized sense strand RNA-seq coverage and spike-in normalized Rpb1 ChIP enrichment over four genes, two with elevated mRNA levels and two with decreased mRNA levels upon Spn1 depletion.

\newpage

\textbf{\large Figure 3}

\includegraphics[width=8.8cm]{figures/figure_3_promoter_swap.pdf}

\vspace{2em}
\textbf{Figure 3. 5$^\prime$ regulatory regions influence Spn1-dependent regulation of transcript levels.}
\textbf{a} Schematic of gene fusions in which the 5$^\prime$ regulatory regions of two Spn1-dependent genes replaced the 5$^\prime$ regulatory region of the YLR454W gene.
The nucleotide coordinates of the native UBI4 and GCV3 regulatory regions are shown, relative to the start of the coding sequence$^{19,81}$.
All fusions were made in the Spn1 degron background.
\textbf{b} Scatterplot summarizing results of the 5$^\prime$ regulatory region replacement experiment.
The y-axis shows RT-qPCR measurements of the change in YLR454W transcript abundance upon Spn1 depletion in the gene fusion strains.
The x-axis shows RNA-seq measurements of the change in transcript abundance upon Spn1 depletion for the native gene whose promoter was used to drive YLR454W expression in the fusion strains.
The line y=x is drawn for comparison.
Error bars indicate 95\% confidence intervals, and points corresponding to the same native gene are slightly jittered by adding a small amount of random noise along the x-axis for clarity.

\newpage

\textbf{\large Figure 4}

\includegraphics[width=18cm]{figures/figure_4_set2_spt6.pdf}

\vspace{2em}
\textbf{Figure 4. Requirement of Spn1 for Spt6 and Set2 association with RNAPII and chromatin.}
\textbf{a} Western blots showing the results of co-immunoprecipitation experiments to analyze the effects of Spn1 or Spt6 depletion on the binding of Spt6, Spn1, and Set2 to Rpb3-FLAG.
Spn1 or Spt6 degron strains (Spn1-AID or Spt6-AID) were treated for 90 minutes with DMSO (lanes marked `D') or 25 $\mu$M IAA (lanes marked `I'), after which Rpb3-FLAG was immunoprecipitated and co-immunoprecipitation of Spn1, Spt6, and Set2 was assayed by western blots using native antibodies.
Rpb3-FLAG levels were measured using anti-FLAG antibody, and an Rpb3-FLAG strain lacking the degron tag on Spn1 or Spt6 was included as an additional non-depleted control.
Shown is one representative western blot of three biological replicates with similar results.
\textbf{b} Barplot showing total levels of Spt6 on chromatin in non-depleted and Spn1-depleted conditions, estimated from ChIP-seq spike-in normalization factors (see Methods).
The values and error bars for each condition indicate the mean $\pm$ standard deviation of two replicates.
\textbf{c} Scatterplot showing changes in Rpb1-normalized Spt6 ChIP enrichment upon Spn1 depletion versus non-depleted Rpb1 enrichment for 5,091 verified coding transcripts.
Rpb1 enrichment values are the relative log$_2$ enrichment of IP over input.
\textbf{d} Average Rpb1-normalized Spt6 ChIP enrichment over 3,087 non-overlapping, verified coding genes aligned by TSS in non-depleted and Spn1-depleted conditions.
The solid line and shading are the median and interquartile range of the ratio calculated from the mean of two Spt6 and four Rpb1 replicates.
\textbf{e} Same as in (b) but for Set2.
\textbf{f} Same as in (c) but for Set2.
\textbf{g} Same as in (d) but for Set2.

\newpage

\textbf{\large Figure 5}

\includegraphics[width=8.8cm]{figures/figure_5_h3.pdf}

\vspace{2em}
\textbf{Figure 5. Histone H3 levels are reduced at the 5$^\prime$ end of a small set of genes after Spn1 depletion.}
\textbf{a} Scatterplot showing change in H3 enrichment upon Spn1 depletion versus non-depleted Rpb1 enrichment for 5,091 verified coding genes.
Rpb1 enrichment values are the relative log$_2$ enrichment of IP over input.
Genes with significant decreases in H3 enrichment are colored red, and a cubic regression spline is overlaid.
\textbf{b} Average H3 ChIP enrichment in non-depleted and Spn1-depleted conditions for the 77 genes with significantly decreased H3 ChIP enrichment over the entire gene upon Spn1 depletion.
The solid line and shading are the median and interquartile range of the mean enrichment over four replicates.
\textbf{c} H3 ChIP enrichment in non-depleted and Spn1-depleted conditions for two genes with significantly decreased H3 ChIP enrichment upon Spn1 depletion.
The mean enrichment over four replicates is shown.

\newpage

\textbf{\large Figure 6}

\includegraphics[width=18cm]{figures/figure_6_h3_mods.pdf}

\vspace{2em}
\textbf{Figure 6. Spn1 is required for normal localization of histone H3 modifications along coding genes.}
\textbf{a} Top: Average H3-normalized H3K36me2 for 3,087 non-overlapping, verified coding genes aligned by TSS in non-depleted and Spn1-depleted conditions.
The solid line and shading are the median and interquartile range of the mean ratio over two replicates.
Vertical dotted lines indicate the distances at which the median signal in each condition reached 90\% of its maximum value.
Bottom: Heatmap of the change in H3-normalized H3K36me2 enrichment upon Spn1 depletion for the same genes, aligned by TSS and arranged by transcript length.
Data is shown for each gene up to 300 bp 3$^\prime$ of the cleavage and polyadenylation site, which is indicated by the white dotted line.
\textbf{b} Same as in (a) but for H3K36me3.
\textbf{c} Same as in (a) but for H3K4me3, with vertical dotted lines instead indicating the distances at which the median signal in each condition dropped to 10\% of its maximum value.

\newpage

\textbf{\large Figure 7}

\includegraphics[width=18cm]{figures/figure_7_splicing.pdf}

\vspace{2em}
\textbf{Figure 7. Depletion of Spn1 results in increased intron retention in ribosomal protein transcripts.}
\textbf{a} Volcano plot showing changes in intron retention upon Spn1 depletion for 252 introns encoded from within coding regions of nuclear genes.
Intron retention is defined as the proportion of unspliced transcripts and was estimated from RNA-seq data (see Methods).
Labeled introns were further assayed by RT-qPCR, as shown in (b).
\textbf{b} RT-qPCR measurements of the ratio of unspliced to spliced transcripts in a wild-type strain treated with DMSO for 90 minutes and the Spn1 degron strain treated with 25 $\mu$M IAA for 90 minutes.
Error bars indicate the mean $\pm$ standard deviation of two replicates.
\textbf{c} Visualization of average changes in H3K36 methylation state at ribosomal protein genes with and without introns.
For each group of genes, the following data are plotted: sense strand RNA-seq coverage, GC\% in a 21-bp window, and H3-normalized H3K36me2 and H3K36me3 ChIP enrichment.
The solid line and shading are the median and interquartile range over the genes in each group.
The introns of RP genes tend to be close to the 5’ end of the gene, the region where RNA-seq signal is depleted (top left panel).

\newpage

\textbf{\large Supplementary Figure 1}

{\includegraphics[width=18cm]{figures/figure_S1_spn1_depletion_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 1. Effects of Spn1 depletion.}
\textbf{a} Viability of a wild-type strain, a strain expressing TIR1 but lacking the degron tag on Spn1 (Spn1-noAID), and the Spn1 degron strain (Spn1-AID) after 90 minutes of treatment with DMSO or 25 $\mu$M IAA, quantified as colony-forming units per OD$_{600}$ unit of culture plated.
Error bars indicate the mean $\pm$ standard deviation of the replicates shown.
\textbf{b} Scatterplots of Spn1 enrichment versus Rpb1 enrichment for 5,091 verified coding genes in non-depleted and Spn1-depleted conditions.
Pearson correlation coefficients are indicated.
Enrichment values are the relative log$_2$ enrichment of IP over input.
\textbf{c} Scatterplot showing Rpb1-normalized Spn1 enrichment versus Rpb1 enrichment for the same genes as in (b) in the non-depleted condition.
Rpb1 enrichment values are the relative log$_2$ enrichment of IP over input.
The Pearson correlation coefficient is indicated, and a linear regression is overlaid.

\newpage

\textbf{\large Supplementary Figure 2}

{\includegraphics[width=18cm]{figures/figure_S2_rnaseq_rpb1_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 2. RNA-seq and RNAPII ChIP-seq.}
\textbf{a} Scatterplot comparing the results of two RNA-seq differential expression analyses for 5,091 verified coding genes.
X-axis values show the fold-change estimated from comparison of the Spn1 degron strain treated with IAA (Spn1-depleted) versus DMSO (non-depleted), the same comparison shown in Fig. 2a.
Y-axis values show the fold-change attributable to the effects of Spn1 depletion, as extracted from a multi-factor analysis incorporating RNA-seq data from additional control strains to be able to separate effects due to Spn1 depletion from effects due to presence of the degron tag on Spn1, expression of degron system component TIR1, or treatment of yeast cells with IAA.
The line y=x is overlaid for comparison.
Results of the simple comparison between Spn1-depleted and non-depleted conditions are used throughout this study for consistency with ChIP-seq analyses for which the multi-factor analysis was not performed.
\textbf{b} Scatterplot comparing changes in transcript abundance upon Spn1 depletion as measured by RT-qPCR and RNA-seq. The line y=x is drawn for comparison, and error bars indicate 95\% confidence intervals.
\textbf{c} Running GSEA enrichment score for genes up- and downregulated during the yeast environmental stress response (ESR)$^{43}$ along genes ranked by change in transcript abundance upon Spn1 depletion as measured by RNA-seq.
The ranks of ESR up- and downregulated genes are marked along the top and bottom of the plot, respectively, and Benjamini-Hochberg adjusted enrichment p-values are indicated.
\textbf{d} Scatterplot comparing changes in transcript abundance upon Spn1 depletion as measured by RNA-seq versus changes in transcript abundance expected from a slow growth signature common to many environmental and genetic perturbations as previously reported$^{44}$.
\textbf{e} Spike-in normalized RNA-seq coverage over four genes with antisense transcripts that are differentially upregulated upon Spn1 depletion.
Sense and antisense strand signal are plotted above and below the x-axis, respectively.
Unlabeled horizontal lines indicate the boundaries of transcripts called by StringTie.
\textbf{f} Barplots showing total levels of Rpb1, Rpb1-Ser5-P, and Rpb1-Ser2-P on chromatin in non-depleted and Spn1-depleted conditions, estimated from ChIP-seq spike-in normalization factors (see Methods).
The values and error bars for each condition indicate the mean $\pm$ standard deviation of the replicates shown.
Note that we observed batch-to-batch variability in Rpb1 levels after Spn1 depletion, with Rpb1 levels decreasing for the two replicates generated from the same chromatin preparations used to ChIP Rpb1-Ser5-P and Rpb1-Ser2-P.

\newpage

\textbf{\large Supplementary Figure 3}

{\includegraphics[width=18cm]{figures/figure_S3_set2_spt6_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 3. Spt6 and Set2 ChIP-seq.}
\textbf{a} Scatterplots showing Spt6 or Set2 enrichment versus Rpb1 enrichment for 5,091 verified coding genes in non-depleted and Spn1-depleted conditions.
Enrichment values are the relative log$_2$ enrichment of IP over input.
Pearson correlation coefficients are indicated.
\textbf{b} Scatterplots showing Rpb1-normalized enrichment of Spt6 or Set2 versus Rpb1 enrichment for the same genes as in (a) in non-depleted and Spn1-depleted conditions.
Pearson correlation coefficients are indicated.
\textbf{c} Representative western blots showing the levels of Spn1, Rpb1, Spt6, and Set2 in the wild type, Spn1-noAID, and Spn1-AID strains treated with DMSO or 25 $\mu$M IAA.
The faint band co-migrating with native Spn1 in the Spn1-AID lanes of the Spn1 blot is non-specific, as it was also present in a \textit{spn1$\Delta$} strain (data not shown).
Pgk1 was used as a loading control.

\newpage

\textbf{\large Supplementary Figure 4}

{\includegraphics[width=18cm]{figures/figure_S4_h3_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 4. Histone H3 ChIP-seq.}
\textbf{a} Average H3 ChIP enrichment for 3,087 non-overlapping, verified coding genes aligned by TSS in non-depleted and Spn1-depleted conditions.
The solid line and shading are the median and interquartile range of the mean enrichment over four replicates.
\textbf{b} Scatterplot showing change in H3 enrichment upon Spn1 depletion over the first 500 bp downstream of the TSS versus non-depleted Rpb1 enrichment over the entire length of 4,924 verified coding genes at least 500 bp long.
Rpb1 enrichment values are the relative log$_2$ enrichment of IP over input.
Genes with significant decreases in H3 enrichment over their first 500 bp are colored red, and a cubic regression spline is overlaid.
\textbf{c} Scatterplots comparing Spn1-dependent changes in H3 enrichment over the entire gene to changes in Rpb1 enrichment, Spt6 enrichment, and Rpb1-normalized Spt6 enrichment.
Values are shown for 5,091 verified coding genes, with genes for which H3 enrichment is significantly reduced upon Spn1 depletion colored red.
A linear regression for each group is overlaid, with Pearson correlation coefficients shown in the bottom right.
\textbf{d} Average H3, Rpb1, Spt6, and Rpb1-normalized Spt6 ChIP enrichment in non-depleted and Spn1-depleted conditions for the 77 genes with significantly reduced H3 enrichment upon Spn1 depletion and for 80 genes without significantly reduced H3 enrichment matched to the first group by expression level and transcript length.
The solid line and shading are the median and interquartile range over the genes in each group.

\newpage

\textbf{\large Supplementary Figure 5}

{\includegraphics[width=18cm]{figures/figure_S5_h3_mods_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 5. Histone H3 modifications ChIP-seq.}
\textbf{a} Representative western blots showing the levels of H3K36me2, H3K36me3, and H3 in the wild type, Spn1-noAID, and Spn1-AID strains treated with DMSO or 25 $\mu$M IAA.
The same Spn1 and Pgk1 panels used in Supplementary Fig. 3c are shown here for reference, as they were generated as part of the same experiment.
\textbf{b} Barplots showing total levels of H3K36me2, H3K36me3, H3K4me3, and H3 on chromatin in non-depleted and Spn1-depleted conditions, estimated from ChIP-seq spike-in normalization factors (see Methods).
The values and error bars for each condition indicate the mean $\pm$ standard deviation of the replicates shown.
\textbf{c} Average H3K36me2, H3K36me3, and H3K4me3 ChIP enrichment in non-depleted and Spn1-depleted conditions for 3,087 non-overlapping, verified coding genes aligned by TSS.
The solid line and shading are the median and interquartile range of the mean enrichment over two replicates.
\textbf{d} Average H3-normalized H3K36me2, H3K36me3, and H3K4me3 ChIP enrichment in non-depleted and Spn1-depleted conditions for 3,087 non-overlapping, verified coding genes separated into tertiles of expression by their non-depleted Rpb1 ChIP enrichment.
The solid line and shading are the median and interquartile range of the mean ratio over two replicates.

\newpage

% \textbf{\large Supplementary Figure 6}

% {\includegraphics[width=18cm]{figure_S6_chd1.pdf}\par}

% \newpage

\textbf{\large Supplementary Figure 6}

{\includegraphics[width=18cm]{figures/figure_S7_splicing_supplemental.pdf}\par}

\vspace{2em}
\textbf{Supplementary Figure 6. Effects of Spn1 depletion at intron-containing ribosomal protein genes.}
\textbf{a} Genomic data visualization for ribosomal protein genes with and without introns.
The solid line and shading for each assay are the median and interquartile range over the genes in each group.
The same sense strand RNA-seq coverage panel used in Fig. 7c is shown here for reference.
\textbf{b} Heatmaps of change upon Spn1 depletion for sense strand RNA-seq signal and H3-normalized H3K36me2 ChIP enrichment over 89 ribosomal protein (RP) genes with introns and 153 non-RP genes with introns.
Genes are aligned by TSS and ordered by distance from the TSS to the midpoint of the first intron.
The 5$^\prime$ and 3$^\prime$ splice sites of each intron are marked with red and orange dots, respectively.

\newpage
\end{document}
